File: crop_quinn.ts
import { Jimp } from 'jimp';

async function cropImage() {
    const path = 'client/public/quinn-knot.png';
    console.log(`Processing ${path}...`);

    try {
        const image = await Jimp.read(path);
        const originalWidth = image.width;
        const originalHeight = image.height;

        console.log(`Original Size: ${originalWidth}x${originalHeight}`);

        // Jimp's autocrop automatically removes same-color borders (transparency)
        image.autocrop();

        console.log(`New Size: ${image.width}x${image.height}`);

        await image.write(path);
        console.log("Successfully cropped and overwritten.");

    } catch (error) {
        console.error("Error:", error);
        process.exit(1);
    }
}

cropImage();

File: crop_quinn.py
from PIL import Image
import os

def crop_transparency(path):
    print(f"Processing {path}...")
    try:
        img = Image.open(path)
        img = img.convert("RGBA")
        
        # Get bounding box of non-zero alpha pixels
        bbox = img.getbbox()
        
        if bbox:
            print(f"Original Size: {img.size}")
            print(f"Cropping to: {bbox}")
            
            cropped = img.crop(bbox)
            print(f"New Size: {cropped.size}")
            
            cropped.save(path)
            print("Successfully cropped and overwritten.")
        else:
            print("Image is fully transparent or empty. No crop needed.")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    target_path = r"c:\Users\n\.gemini\antigravity\playground\blazing-aldrin\client\public\quinn-knot.png"
    crop_transparency(target_path)
